framework/src/main/java/org/checkerframework/common/aliasing/AliasingTransfer.java

```diff
@@ -83,41 +89,80 @@ public TransferResult<CFValue, CFStore> visitAssignment(
      */
     @Override
     protected void processPostconditions(
-            MethodInvocationNode n, CFStore store, ExecutableElement methodElement, Tree tree) {
-        super.processPostconditions(n, store, methodElement, tree);
-        if (TreeUtils.isEnumSuper(n.getTree())) {
+            Node n, CFStore store, ExecutableElement executableElement, ExpressionTree tree) {
+        MethodInvocationNode methodInvocationNode = null;
+        ObjectCreationNode objectCreationNode = null;
+        if (n instanceof MethodInvocationNode) {
+            methodInvocationNode = (MethodInvocationNode) n;
+        } else if (n instanceof ObjectCreationNode) {
+            objectCreationNode = (ObjectCreationNode) n;
+        } else {
+            throw new BugInCF(
+                    "processPostconditions in AliasingTransfer expects "
+                            + "a MethodInvocationNode or ObjectCreationNode argument; received a "
+                            + n.getClass().getSimpleName());
+        }
+        super.processPostconditions(n, store, executableElement, tree);
+        if (methodInvocationNode != null && TreeUtils.isEnumSuper(methodInvocationNode.getTree())) {
             // Skipping the init() method for enums.
             return;
         }
-        List<Node> args = n.getArguments();
-        List<? extends VariableElement> params = methodElement.getParameters();
+        List<Node> args =
+                methodInvocationNode != null
+                        ? methodInvocationNode.getArguments()
+                        : objectCreationNode.getArguments();
+        List<AnnotatedTypeMirror> params = null;
+        AnnotatedExecutableType annotatedType = factory.getAnnotatedType(executableElement);
+        if (methodInvocationNode != null) {
+            params = annotatedType.getParameterTypes();
+        } else if (objectCreationNode != null) {
+            AnnotatedTypeFactory.ParameterizedExecutableType fromUse =
+                    factory.constructorFromUse((NewClassTree) tree);
+            AnnotatedExecutableType constructorType = fromUse.executableType;
+            params =
+                    AnnotatedTypes.adaptParameters(
+                            factory, constructorType, ((NewClassTree) tree).getArguments());
+            List<Node> tmp = new ArrayList<>();
+            // expand ArrayCreationNode such as new int []
```

Comment:
What is this about and why is it only needed here and only for the Aliasing Checker?

Reply:
When passing varargs into a constructor, the args are like `new int [] {1, 2,3}`, I think we need to expand the parameters... 

Only Aliasing Checker has such logic that remove specific args in the store when calling processPostcondition...